.github/actions/lint-front-matter/main.js

```diff
@@ -0,0 +1,33 @@
+const core = require('@actions/core');
+const lintAllFiles = require('./linters/all-files');
+const lintAddedFiles = require('./linters/added-files');
+const lintModifiedFiles = require('./linters/modified-files');
+const combineByFile = require('./utils/combine-by-file');
+const logOutput = require('./utils/log-output');
+
+async function run() {
+  try {
+    // Inputs come in as a space separated list of file paths.
+    const added = core.getInput('added').length
+      ? core.getInput('added').split(' ')
+      : [];
+    const modified = core.getInput('modified').length
+      ? core.getInput('modified').split(' ')
+      : [];
+    const all = [...added, ...modified];
+
+    const results = [
+      ...(await lintAllFiles(all)),
```

Comment:
I really think this would be much simpler if these methods just existed in this file. The three other files end up with tons of boilerplate. It looks scary to add another one since that seems fragile.

Reply:
I'm aware this ties into how you test them. But I really think each _group_ of tests could look like:

```js
const allFilesRun = (frontMatter) => {
  return [
    hasProperty.test(frontMatter, 'date'),
    hasAltText.test(frontMatter),
  ];
};

// maybe later
module.exports.allFilesRun = allFilesRun;
```